recipe/meta.yaml

```diff
@@ -6,49 +6,79 @@ package:
   version: {{ version }}
 
 source:
-  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/model-index-{{ version }}.tar.gz
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
     sha256: 2f9870200f3b00813881b07b90d2c67291663534e43915c75fe476b6977bf2ad
+    patches:
+      - 0001-write-deps-from-missing-file.patch
   - url: https://raw.githubusercontent.com/paperswithcode/model-index/main/LICENSE
     sha256: addec7ad58188c4d6df6f578fb0c6877bd799d787ecbd9f133213de995b345a4
-  - url: https://raw.githubusercontent.com/paperswithcode/model-index/main/requirements.txt
-    sha256: 8356928a01bcda562a9d2a8168997db018e4c803b43ad9c6d1e137f80ca95759
+  - git_url: https://github.com/paperswithcode/model-index
+    git_rev: 09c7865ab2cbe00827d3af879a2faf7d286cb1b7
+    folder: github_repo
 
 build:
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
   number: 0
   entry_points:
     - mi = modelindex.__main__:cli
 
 requirements:
+  build:
+    - patch  # [not win]
+    - m2-patch  # [win]
+    - git  # [not win]
   host:
-    - python >=3.7
+    - python
     - pip
     - pyyaml
```

Comment:
are these next four requirements necessary in the host section? I know it's in the requirements.txt, which is usually for setting up dev environments, but the fact that upstream takes the runtime requirements from requirements.txt makes me wonder whether they're required for building.